src/lib/paymentToken.ts

```diff
@@ -0,0 +1,43 @@
+import { ConvertedTokenInfo } from '@zero-tech/zns-sdk';
+import * as ZNS from '@zero-tech/zns-sdk';
+
+interface PaymentTokenCache {
+	[index: string]: ConvertedTokenInfo;
+}
+
+var paymentTokenCache: PaymentTokenCache = {};
+const IS_TEST_MODE = true;
+
+const getPaymentTokenInfo = async (
+	sdk: ZNS.Instance,
+	tokenId: string,
+): Promise<ConvertedTokenInfo> => {
```

Comment:
nit: could be nice to turn this into a hook instead of having to send the SDK instance, but this works too.

Reply:
Can't use a hook as we need to fetch data for multiple domains and probably in a loop.